deploy/bin/build-push-edge-endpoint-image.sh

```diff
@@ -37,8 +50,10 @@ aws ecr get-login-password --region us-west-2 | docker login \
                   --username AWS \
                   --password-stdin  ${ECR_URL}
 
-# Check if the first argument is "dev". If it is, only build the image for the current platform
+# Check if the first argument is "dev". If it is, only build the image for the current platform and don't push the
+# :latest tag.
 if [ "$1" == "dev" ]; then
+  echo "Received argument 'dev'. Building for current platform only. Latest tag will not be pushed."
   docker build --tag ${EDGE_ENDPOINT_IMAGE} ../..
   docker tag ${EDGE_ENDPOINT_IMAGE}:latest ${ECR_URL}/${EDGE_ENDPOINT_IMAGE}:${TAG}
   docker push ${ECR_URL}/${EDGE_ENDPOINT_IMAGE}:${TAG}
```

Comment:
Since we are proposing using `dev` tag for safer testing, I think we should also make sure that we can build the image for both ARM and x86 so that people testing it out with ARM devices can also run the `dev` images.

We could just move this portion after buildx command use it:
```
# Build image for amd64 and arm64
docker buildx build \
  --platform linux/arm64,linux/amd64 \
  --tag ${ECR_URL}/${EDGE_ENDPOINT_IMAGE}:${TAG} \
  ../.. --push
```